### DISCUSSION TITLE
Add button missing for Secrets in UI

### DISCUSSION THREAD

**Discussion by lancamat1** [Onemind-Services-LLC/netbox-secrets#166] [Q&A] [answered]
I may be stupid to spend all day today figuring out how to add Secrets via the web UI, but i cannot find the add button for secrets anywhere. Seems like the most basic task and yet I am the only one not figuring it out and having to ask here.

The usual "netbox" way from the list view is not it, and this seems to be by design:
class SecretListView(generic.ObjectListView):
...
    actions = {
        "export": {"view"},
        "bulk_edit": {"change"},
        "bulk_delete": {"delete"},
    }
Documentation does not touch on this either.
How does the expected workflow with adding the secrets look like?

My configuration is nothing special, modified the defaults to enable assigning secrets to VirtualChassis:
    'netbox_secrets': {
        'apps': ['dcim.device','dcim.VirtualChassis'],
        'top_level_menu': True,
    }
    
Versions are 2.1.0 for the plugin and 4.1.2 for netbox

<img width="1159" alt="Screenshot 2024-10-23 at 1 13 56" src="https://github.com/user-attachments/assets/3492b84e-1bfb-4848-be78-463faa8bce95">

**Comment by kprince28** [comment]
You can find the `Add Secret` button within the apps you've defined. For example, if you've added `dcim.device` in your plugin configuration, follow these steps:

- Navigate to the specific device by selecting it in your app.
- Click to open the device details.
- In the device's object view, you'll see the "Add Secret" button.

**Reply by lancamat1** [reply]
Thank you for your reply, somehow I did not notice this on the device.
Is VirtualChassis supported too? I have it listed in apps, but the secrets is not visible on detail view.

**Reply by kprince28** [reply, marked as answer]
The issue is due to an incorrect configuration. It should be defined as:
`'apps': ['dcim.device', 'dcim.virtualchassis']`.
